femwell/maxwell/waveguide.py

```diff
@@ -330,7 +369,10 @@ def compute_modes(
     if order == 1:
         element = ElementTriN1() * ElementTriP1()
     elif order == 2:
-        element = ElementTriN2() * ElementTriP2()
+        # element = ElementTriN2() * ElementTriP2()
+        from skfem.element import ElementQuad2
+
+        element = ElementTriN2() * ElementQuad2  # ElementTriP2()
```

Comment:
This should stay as-is, we don't have yet support for Quad-meshes yet (and we won't before we decide to add tests, but that's mostly depending on when meshwell will be able to reliably produce quad meshes).

In case we want to add it, we'd need to check there if the mesh is tri or quad and select the element accordingly (or even better, have two elements and use the adequate one for the mesh element)

For now we can just stay with Tri

```suggestion
        element = ElementTriN2() * ElementTriP2()
```

Reply:
Yeah, you are definitely right, it completely makes sense to let it as it was.